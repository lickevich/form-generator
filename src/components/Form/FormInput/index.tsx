import React from "react";

import { IInput } from "../../../type";
import Row from "../Row";
import Col from "../Col";

const FormInput: React.FC<IInput> = ({ index, label, ...rest }) => {
  const id = `${label}${index}`;

  return (
    <Row>
      <Col>
        <label htmlFor={id} title={label}>
          {label}
        </label>
      </Col>
      <Col>
        <input id={id} {...rest} />
      </Col>
    </Row>
  );
};

export default FormInput;
