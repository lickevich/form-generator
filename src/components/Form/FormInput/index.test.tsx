import React from "react";
import { render } from "@testing-library/react";

import FormInput from "./";

describe("FormInput", () => {
  it("render the component", () => {
    const component = (
      <FormInput index="0" type="text" label="Label_input" autoComplete="on" />
    );
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
