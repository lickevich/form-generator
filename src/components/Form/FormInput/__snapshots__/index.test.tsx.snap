// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa kTYFsi"
  >
    <div
      class="sc-gsDJrp fhKCEr"
    >
      <label
        for="Label_input0"
        title="Label_input"
      >
        Label_input
      </label>
    </div>
    <div
      class="sc-gsDJrp fhKCEr"
    >
      <input
        autocomplete="on"
        id="Label_input0"
        type="text"
      />
    </div>
  </div>
</DocumentFragment>
`;
